### DISCUSSION TITLE
Having the default locale in the URL too?

### DISCUSSION THREAD

**Discussion by ygini** [facebook/docusaurus#9918] [Q&A]
Hello,

My docusaurus website is set to be english by default and propose content for other language (french for example) too.

Currently the URL scheme is for the ressources "test" to be at `/test` for english and `/fr/test` for french.

Is there a way to have explicit URL for english too (`/en/test`) and to have no language selection (`/test`) triggering the locale detection to redirect either to `/fr/test` or `/en/test`?

This would allow us to have link in our product set as `/test` and let docusaurus handle the explicit choice of langage base on user preference.

**Comment by slorber** [comment]
You can set the default language as `/en/` instead of `/` through i18n config options

We don't provide a locale detection system, we don't have a server, and client redirects are not ideal, but you can use some kind of edge redirection middleware through your CDN host if it supports it (Vercel, Netlify, Cloudflare supports this)
